homeassistant/components/nasweb/__init__.py

```diff
@@ -0,0 +1,120 @@
+"""The NASweb integration."""
+
+from __future__ import annotations
+
+import logging
+
+from webio_api import WebioAPI
+from webio_api.api_client import AuthError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.network import NoURLAvailableError
+from homeassistant.util.hass_dict import HassKey
+
+from .const import DOMAIN, MANUFACTURER, SUPPORT_EMAIL
+from .coordinator import NASwebCoordinator
+from .nasweb_data import NASwebData
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+NASWEB_CONFIG_URL = "https://{host}/page"
+
+_LOGGER = logging.getLogger(__name__)
+type NASwebConfigEntry = ConfigEntry[NASwebCoordinator]
+DATA_NASWEB: HassKey[NASwebData] = HassKey(DOMAIN)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: NASwebConfigEntry) -> bool:
```

Comment:
During setup, we should check that the serial number of the device we're connecting to is matching the stored serial number (the config entry's unique ID), in case the device's IP address has changed.